### PR TITLE
Make the 5 second timeout on test-running specifiable by an optional parameter to `phantom.test.add`

### PR DESCRIPTION
**Pull request by harob**
Note that because this is an optional parameter it should not break any existing tests.
Also, this makes `phantom.test.add` mirror `@body.assertFirst`, as both now optionally take the optional argument `total`.

Please let me know if there are any edits you would like me to make!

### CODE REVIEW

**Comment by joshbuddy** [comment]
A couple of questions about this .. one, where is it needed? This timer gets destroy and recreated between each assertion. I'm wondering where you've bumped against it. Two, wanna re-merge and write some tests? Thank you!

**Comment by harob** [comment]
I need this functionality to test a page with a Flash application that takes a very long time (~10 seconds) to initialize.

So my test looks something like the following:

```
phantom.test.add "Long running test", total: 20, ->
  @get '/page_with_flash_app', ->
    @wait 15, (->
      @body.assertFirst '#flashApp', (f) ->
        f.someFlashAppAPIMethod() == "expectedValue"
      @succeed()
    )
```

If I missing some existing functionality that already accomplishes this, please let me know!

Regarding adding tests: It looks like the /test directory contains sample apps with sample tests; so should I just add a sample test that uses this new functionality? Or is there some automated unit testing of the framework that I am missing?

**Comment by joshbuddy** [comment]
Lemme clean up the how-to on testing here. Go ahead and re-merge from master so that this merges cleanly.

**Comment by harob** [comment]
I merged upstream, so this should presumably now merge cleanly. Please me know when you want me to update the tests.

Great work on this whole framework by the way!

File: test/working_ghost/test_ghostmore.coffee [hunk 1]
```diff
@@ -16,7 +16,7 @@ phantom.test.add "Simple form with wait", ->
     @wait 1, ->
       @succeed()
       
-phantom.test.add "Slow form", ->
```
Comment (joshbuddy) [inline test/working_ghost/test_ghostmore.coffee:19]: Why change this test? Shouldn't you add a different test for this? Also, can you add the negative test case to non_working_ghost?

Comment (harob) [inline test/working_ghost/test_ghostmore.coffee:19]: Will do.
